### PR TITLE
Fixes #12237: Add examples in documentation of condition_from_expression 

### PR DESCRIPTION
**Pull request by ncharles**
https://www.rudder-project.org/redmine/issues/12237

### CODE REVIEW

File: tree/30_generic_methods/condition_from_command.cf [hunk 1]
```diff
@@ -18,16 +18,31 @@
 
 # @name Condition from command 
 # @description Execute a command and create outcome classes depending on its exit code
-# @documentation This bundle will define a class `condition_from_command_${condition_prefix}_{kept,not_ok,ok,reached}`
-# depending on the exit codes given in parameters.
-#
-# This bundle will additionally produce a `${condition_prefix}_true` or a `${condition_prefix}_false` condition depending on the result on the command:
+# @documentation This method executes a command, and defines a `${condition_prefix}_true` or a `${condition_prefix}_false` condition depending on the result on the command:
```
Comment (amousset) [inline tree/30_generic_methods/condition_from_command.cf]: *of* the command

**Comment by ncharles** [comment]
Commit modified

File: tree/30_generic_methods/condition_from_expression.cf [hunk 2]
```diff
@@ -18,16 +18,29 @@
 
 # @name Condition from expression
 # @description Create a new condition class
-# @documentation This bundle will define a class `condition_from_expression_${condition_prefix}_{kept,ok,reached}`
-#
-# This bundle will additionally produce a `${condition_prefix}_true` or a `${condition_prefix}_false` condition depending on the result on the expression:
+# @documentation This method evaluates an expression, and produce a `${condition_prefix}_true` or a `${condition_prefix}_false` condition depending on the result on the expression:
```
Comment (amousset) [inline tree/30_generic_methods/condition_from_expression.cf]: produces

File: tree/30_generic_methods/condition_from_expression_persistent.cf [hunk 3]
```diff
@@ -18,16 +18,31 @@
 
 # @name Condition from expression persistent
 # @description Create a new condition class that persists across runs
-# @documentation This bundle will define a class `condition_from_expression_persistent_${condition_prefix}_{kept,ok,reached}`
-#
-# This bundle will additionally produce a `${condition_prefix}_true` or a `${condition_prefix}_false` condition depending on the result on the expression:
+# @documentation This method evaluates an expression, and produce a `${condition_prefix}_true` or a `${condition_prefix}_false` condition depending on the result on the expression, for the `duration` time:
```
Comment (amousset) [inline tree/30_generic_methods/condition_from_expression_persistent.cf]: same here

File: tree/30_generic_methods/condition_from_expression_persistent.cf [hunk 4]
```diff
@@ -18,16 +18,31 @@
...
 #
 # * If the expression results in a "defined" state, this will produce a kept outcome class and a `${condition_prefix}_true` condition
 # * If the expression results in an "undefined" state, this will produce a kept outcome class and a `${condition_prefix}_false` condition
 #
-# The created condition (class in cfengine speaking) is global to the agent and is persisted across runs.
+# Calling this method with a condition expression transforms a complex expression into a single class condition.
+#
+# The created condition is global to the agentand is persisted across runs.
```
Comment (amousset) [inline tree/30_generic_methods/condition_from_expression_persistent.cf]: missing space

Comment (amousset) [reply tree/30_generic_methods/condition_from_expression_persistent.cf]: could you also explain what persistence is, I don't think it is known by Rudder users.

**Comment by ncharles** [comment]
Commit modified

**Comment by amousset** [comment]
This is great!

**Comment by ncharles** [comment]
OK, merging this PR

**Comment by ncharles** [comment]
OK, merging this PR

**Pull request merged**